### DISCUSSION TITLE
Automatic parent/child phase selection in `setParentGrainReconstructor`

### DISCUSSION THREAD

**Discussion by jlhostis** [ORTools4MTEX/ORTools#33] [Q&A] [answered]
Hi _ORTools_ community, 

for the purpose of optimisation and batch processing of a large number of EBSD maps, I would like to automatically assign the *parent* and *child* attributes to the phases when doing the PAG reconstruction.

Indeed, when using the command `job = setParentGrainReconstructor(ebsd,grains);`, I am prompted with a window to select which phase will be parent and which phase will be child. As I am always clicking the `Martensite` and `Austenite` buttons, **is there a way to automatically select the phases ?**

And if this is not existing yet, could it be requested as a feature, by passing `'parentPhase', 'Austenite'` as an argument to the `setParentGrainReconstructor` function for example.

Thank you in advance, 

Sincerely, 
Jules

**Comment by Deep0161** [comment, marked as answer]
If you just initialise your _parentGrainReconstructor_ class with _parentGrainReconstructor(ebsd, grains)._ It invokes and automatically assigns the larger fraction phase of your original ebsd data as the parent phase and the other phase as child. So I think even if you don't run the assign phase function then also, your code should work alright.

Cheers
Deep

**Comment by jlhostis** [comment]
Indeed thank you @Deep0161 for your swiftness !

I did not know I could use the function `parentGrainReconstructor` directly. I had an issue where I have more than my parent aud child phases present in the dataset (carbides), but I can force `job.csParent = parentCS` where `parentCS` is of class `@crystalSymmetry`.

Thank you again,
Jules
